region/tourian/main/Blue Hopper Room.json

```diff
@@ -603,64 +627,89 @@
     },
     {
       "link": [2, 2],
-      "name": "G-Mode Setup - Tank During Setup",
-      "notable": false,
+      "name": "G-Mode Setup",
+      "requires": [],
+      "exitCondition": {
+        "leaveWithGModeSetup": {}
+      },
+      "flashSuitChecked": true,
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
+    },
+    {
+      "link": [2, 2],
+      "name": "G-Mode Regain Mobility",
       "requires": [
         {"enemyDamage": {
           "enemy": "Blue Sidehopper",
           "type": "contact",
           "hits": 1
         }}
       ],
-      "exitCondition": {
-        "leaveWithGModeSetup": {}
-      },
-      "unlocksDoors": [{"types": ["ammo"], "requires": []}],
+      "gModeRegainMobility": {},
       "flashSuitChecked": true
     },
     {
-      "link": [2, 2],
-      "name": "G-Mode Setup - Prepared Dodge",
-      "notable": false,
+      "link": [3, 2],
+      "name": "Sidehopper Hit on Entry",
+      "requires": [
+        {"enemyDamage": {
+          "enemy": "Blue Sidehopper",
+          "type": "contact",
+          "hits": 1
+        }}
+      ],
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Sidehopper Prepared Screw Dodge",
       "requires": [
         "canPrepareForNextRoom",
-        {"or": [
-          {"and": [
-            "ScrewAttack",
-            "SpaceJump"
-          ]},
-          {"and": [
-            "Morph",
-            {"or": [
-              "canTrickyJump",
-              {"enemyKill": {
-                "enemies": [["Blue Sidehopper"]]
-              }}
-            ]}
-          ]}
-        ]}
+        "ScrewAttack",
+        "SpaceJump",
+        "canCarefulJump"
+      ],
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Sidehopper Prepared Morph Dodge",
+      "requires": [
+        "canPrepareForNextRoom",
+        "Morph",
+        "canTrickyJump"
       ],
-      "exitCondition": {
-        "leaveWithGModeSetup": {}
-      },
       "unlocksDoors": [
         {"types": ["super"], "requires": []},
         {"types": ["missiles", "powerbomb"], "requires": ["never"]}
       ],
-      "flashSuitChecked": true
+      "note": "Enter the room morphed to avoid a hopper hit on entry."
```

Comment:
```suggestion
      "note": [
        "Enter the room morphed to avoid a hopper hit on entry.",
        "Roll under the bottom hopper to lead it to the right, and then roll back to reach the door safely.
      ]
```